agent/consul/prepared_query_endpoint.go

```diff
@@ -267,6 +267,43 @@ func (p *PreparedQuery) Get(args *structs.PreparedQuerySpecificRequest,
 		})
 }
 
+// GetByExactName returns a single prepared query by Name.
+func (p *PreparedQuery) GetByExactName(args *structs.PreparedQuerySpecificRequest,
+	reply *structs.IndexedPreparedQueries) error {
+	if done, err := p.srv.ForwardRPC("PreparedQuery.GetByExactName", args, reply); done {
+		return err
+	}
+
```

Comment:
So I think we want to check for PreparedQueryRead permission here.

The setup logic for the auth could look similar to this:
https://github.com/hashicorp/consul/blob/8a6e75ac815a5bdfd9fdbd3528c985d5fdde9c1b/agent/consul/acl_endpoint.go#L1411

Reply:
Thank you for the example 👍 . I attempted to check the PreparedQueryRead permission in https://github.com/hashicorp/consul/pull/12053/commits/a08da226c1c4bb835821e5e3b49c812e9eb75d34. Please review it again when you have time.